internal/ethapi/api.go

```diff
@@ -615,9 +608,9 @@ func (s *PublicBlockChainAPI) doCall(ctx context.Context, args CallArgs, blockNr
 			return "0x", gas, err
 		}
 		return common.ToHex(res), gas, err
+	} else {
+		return "0x", common.Big0, err
 	}
```

Comment:
There's no need for this else branch here. The if branch already returns. Please revert to the original code.